src/app/schemas/productSchema.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from "zod";
+import { AvailabilityStatus, Category, returnPolicy } from "@/types/product";
+
+export const productSchema = z.object({
```

Comment:
I'd suggest reusing this schema everywhere where you need a product schema. I  think I saw another declaration of it in the action files.